File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.py
def solution(num_list):
    answer = 0
    for n in num_list:
        while n > 1:  # n이 1이 될 때까지 반복
            if n % 2 == 0:  # 짝수일 경우
                n //= 2
            else:  # 홀수일 경우
                n = (n - 1) // 2
            answer += 1  # 연산 횟수 추가
    return answer

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        int temp = 0;
        int count = 0;
        
        for (int i = 0; i < num_list.length; i++) {
            temp = num_list[i];
            while (temp != 1) {
                if (temp % 2 == 0) {
                    temp = temp / 2;
                }
                else if (temp % 2 == 1) {
                    temp = (temp - 1) / 2;
                }
                answer++;
    
            }
        }
        return answer;
    }
}